clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2705,6 +2705,30 @@ An error will be given if:
   }];
 }
 
+def AMDGPUNumWorkGroupsDocs : Documentation {
+  let Category = DocCatAMDGPUAttributes;
+  let Content = [{
+The number of work groups specifies the number of work groups when the kernel
+is dispatched.
+
+Clang supports the
+``__attribute__((amdgpu_num_work_groups(<x>, <y>, <z>)))`` attribute for the
+AMDGPU target. This attribute may be attached to a kernel function definition
+and is an optimization hint.
+
+``<x>`` parameter specifies the maximum number of work groups in the x dimentsion.
```

Comment:
```suggestion
``<x>`` parameter specifies the maximum number of work groups in the x dimension.
```

Also, we should be more clear/elaborate more what `x`, `y`, and `z` dimensions mean here. One thing I note is that `OpenCL` (IIRC?) actually reverses these?  So it is VERY important that we document both order and meaning explicitly.

Reply:
The 3 numbers specify the 3 dimensions for a 3D grid of threads. Conventionally the dimensions are referred to as x, y, and z. See attributes such as `reqd_work_group_size`.